cmd/k8s-deployment-scaler/main.go

```diff
@@ -25,26 +28,22 @@ import (
 // Define the global clientset variable using kubernetes.Interface
 var clientset kubernetes.Interface
 
-// DeploymentInfo stores the relevant information about a Deployment
-type DeploymentInfo struct {
-	Replicas   int32
-	Deployment *appsv1.Deployment
-}
+var (
+	deploymentLister  appslisters.DeploymentLister
+	deploymentsSynced cache.InformerSynced
+)
 
-var deploymentInformer cache.SharedIndexInformer
+func getKubernetesConfig() (*rest.Config, error) {
+	return clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
+}
 
 // main initializes and runs the server, setting up TLS configuration, handling graceful shutdown, and defining HTTP routes.
 func main() {
 	// Initialize Kubernetes client
 	var config *rest.Config
 	var err error
 
-	if os.Getenv("KUBERNETES_SERVICE_HOST") != "" {
-		config, err = rest.InClusterConfig()
-	} else {
-		config, err = clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
-	}
-
+	config, err = getKubernetesConfig()
```

Comment:
```suggestion
	config, err := getKubernetesConfig()
```

and remove the vars above. Looks leftover from earlier changes.

Reply:
Corrected. Thank you.